src/xhydro/extreme_value_analysis/parameterestimation.py

```diff
@@ -0,0 +1,526 @@
+    shapecov : list of Variable
+        List of variables to be used as covariates for the shape parameter.
+    niter : int
+        Number of iterations for the Bayesian sampler.
+    warmup : int
+        Number of warmup iterations for the Bayesian sampler.
+
+    Returns
+    -------
+    list
+        An array of fitted distribution parameters and their confidence intervals at 95% confidence.
+    """
+    jl_y = py_list_to_jl_vector(y)
+    jl_logscalecov, jl_shapecov = jl_variable_fit_parameters([logscalecov, shapecov])
+    jl_model = Extremes.gpfitbayes(
+        jl_y,
+        logscalecov=jl_logscalecov,
+        shapecov=jl_shapecov,
+        niter=niter,
+        warmup=warmup,
+    )
+    return _param_cint(jl_model, bayesian=True)
+
+
+METHOD_NAMES = {
+    "ML": "maximum likelihood",
+    "PWM": "probability weighted moments",
+    "BAYES": "bayesian",
+}
+
+DIST_NAMES = {
+    "genextreme": "<class 'scipy.stats._continuous_distns.genextreme_gen'>",
+    "gumbel_r": "<class 'scipy.stats._continuous_distns.gumbel_r_gen'>",
+    "genpareto": "<class 'scipy.stats._continuous_distns.genpareto_gen'>",
+}
+
+
+def fit(
+    ds: xr.Dataset,
+    dist: str | scipy.stats.rv_continuous = "genextreme",
+    method: str = "ML",
+    dim: str = "time",
+) -> xr.Dataset:
+    r"""Fit an array to a univariate distribution along the time dimension.
+
+    Parameters
+    ----------
+    ds : xr.DataSet
+        Time series to be fitted along the time dimension.
+    dist : {"genextreme", "gumbel_r", "genpareto"} or rv_continuous distribution object
+        Name of the univariate distributionor the distribution object itself.
+        Supported distributions are genextreme, gumbel_r, genpareto.
+    method : {"ML", "PWM", "BAYES}
+        Fitting method, either maximum likelihood (ML), probability weighted moments (PWM) or bayesian (BAYES).
+        The PWM method is usually more robust to outliers.
+    dim : str
+        The dimension upon which to perform the indexing (default: "time").
+
+    Returns
+    -------
+    xr.DataArray
+        An array of fitted distribution parameters.
+
+    Notes
+    -----
+    Coordinates for which all values are NaNs will be dropped before fitting the distribution. If the array still
+    contains NaNs or has less valid values than the number of parameters for that distribution,
+    the distribution parameters will be returned as NaNs.
+    """
+    method = method.upper()
+    _check_fit_params(dist, method)
+    dist_params = _get_params(dist)
+    dist = get_dist(dist)
+
+    out = []
+    data = xr.apply_ufunc(
+        _fitfunc_1d,
+        ds,
+        input_core_dims=[[dim]],
+        output_core_dims=[["dparams"]],
+        vectorize=True,
+        dask="parallelized",
+        output_dtypes=[float],
+        keep_attrs=True,
+        kwargs=dict(
+            # Don't know how APP should be included, this works for now
```

Comment:
Isn't APP a `method`? In any case, it usually gives poor results. I think we can skip it.

Reply:
this was already here, left it there